### PR TITLE
Fix deferred options

### PR DESCRIPTION
**Pull request by mponomar**
Fixes legacy_defaults

### CODE REVIEW

File: db/config.c [hunk 1]
```diff
@@ -234,24 +234,27 @@ static int lrl_if(char **tok_inout, char *line, int line_len, int *st,
     return 1; /* there was no "if" statement or it was true */
 }
 
-int process_deferred_options(struct dbenv *dbenv,
```
Comment (nirbhayc) [inline db/config.c]: I'd keep this and check the return code of process_command() in deferred_do_commands() and pre_read_option() in pre_read_deferred_callback(), instead.

File: db/config.c [hunk 2]
```diff
@@ -403,16 +409,9 @@ struct dbenv *read_lrl_file_int(struct dbenv *dbenv, const char *lrlname,
         read_lrl_option(dbenv, line, &options, strlen(line));
     }
     options.lineno = 0;
-    rc = process_deferred_options(dbenv, DEFERRED_LEGACY_DEFAULTS, &options,
+    process_deferred_options(dbenv, DEFERRED_LEGACY_DEFAULTS, &options,
                                   read_lrl_option);
-    if (rc) {
```
Comment (nirbhayc) [inline db/config.c]: Check rc and abort on error?

File: db/db_tunables.h [hunk 3]
```diff
@@ -672,7 +672,7 @@ REGISTER_TUNABLE("move_deadlock_max_attempt", NULL, TUNABLE_INTEGER,
 REGISTER_TUNABLE("name", NULL, TUNABLE_STRING, &name, DEPRECATED | READONLY,
                  NULL, NULL, NULL, NULL);
 REGISTER_TUNABLE("natural_types", "Same as 'nosurprise'", TUNABLE_BOOLEAN,
-                 &gbl_surprise, READONLY | NOARG, NULL, NULL, NULL, NULL);
+                 &gbl_surprise, INVERSE_VALUE | READONLY | NOARG, NULL, NULL, NULL, NULL);
```
Comment (nirbhayc) [inline db/db_tunables.h]: Sorry :(

File: db/db_tunables.h [hunk 4]
```diff
@@ -775,7 +775,7 @@ REGISTER_TUNABLE("no_static_tag_blob_fix", NULL, TUNABLE_BOOLEAN,
                  &gbl_force_notnull_static_tag_blobs,
                  INVERSE_VALUE | READONLY | NOARG, NULL, NULL, NULL, NULL);
 REGISTER_TUNABLE("nosurprise", NULL, TUNABLE_BOOLEAN, &gbl_surprise,
-                 INVERSE_VALUE | READONLY | NOARG, NULL, NULL, NULL, NULL);
+                 READONLY | NOARG, NULL, NULL, NULL, NULL);
```
Comment (nirbhayc) [inline db/db_tunables.h]: I think we need the INVERSE_VALUE flag here, "nosurprise" ==> !"surprise" ==> !gbl_surprise

File: db/db_tunables.h [hunk 5]
```diff
@@ -1067,7 +1067,7 @@ REGISTER_TUNABLE("static_tag_blob_fix", NULL, TUNABLE_BOOLEAN,
                  &gbl_force_notnull_static_tag_blobs, READONLY | NOARG, NULL,
                  NULL, NULL, NULL);
 REGISTER_TUNABLE("surprise", NULL, TUNABLE_BOOLEAN, &gbl_surprise,
-                 READONLY | NOARG, NULL, NULL, NULL, NULL);
+                 INVERSE_VALUE | READONLY | NOARG, NULL, NULL, NULL, NULL);
```
Comment (nirbhayc) [inline db/db_tunables.h]: "surprise" ==> gbl_surprise

File: db/db_tunables.h [hunk 6]
```diff
@@ -1107,7 +1107,7 @@ REGISTER_TUNABLE("track_berk_locks", NULL, TUNABLE_INTEGER,
 REGISTER_TUNABLE("udp", NULL, TUNABLE_BOOLEAN, &gbl_udp, READONLY | NOARG, NULL,
                  NULL, NULL, NULL);
 REGISTER_TUNABLE("unnatural_types", "Same as 'surprise'", TUNABLE_BOOLEAN,
-                 &gbl_surprise, INVERSE_VALUE | READONLY | NOARG, NULL, NULL,
+                 &gbl_surprise, READONLY | NOARG, NULL, NULL,
```
Comment (nirbhayc) [inline db/db_tunables.h]: okay

**Comment by ldennis** [comment]
see #798

**Pull request closed without merging**